packages/components/script.tsx

```diff
@@ -64,6 +65,11 @@ const Script: React.FC<ScriptProps> = ({ src, code, id }) => {
       // Add the ID if specified.
       if (id) script.id = id;
 
+      // Add any other props to the internal <script> tag.
+      for (let key of Object.keys(props)) {
+        script[key] = props[key];
```

Comment:
This could fail when assigning event handlers and data attributes.

My guess is that using [`Element.setAttribute(key, value)`](https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute) here would be a better option, as it would:

- lowercase attributes automatically (i.e `crossOrigin` → `crossorigin`)
- handle data attributes:
  ```js
  script.setAttribute("data-ad-client", "ca-pub-123");
  script.dataset.adClient; // This returns "ca-pub-123".

The only pitfall is that `className` would need to be handled apart.

```suggestion
        script.setAttribute(key === "className" ? "class" : key, props[key]);
```

Reply:
There might be a bit more to it other than a straight lowercase and `setProperty`. For example functions would need to be assigned. Think of `onLoad` would need to be assigned to `onload` and so on.

There's a way to handle both of them like this:

```js
for (let key in props) {
  const value = props[key];

  // If the current key exists in the `dom` interface
  // we can assign the value.
  if (key in script) {
    // If this is an event handler, lowercase the key
    if (/^on/g.test(key)) {
      key = key.toLowerCase();
    }
    script[key] = value;
  } else if(typeof value !== 'function' && typeof value !== 'object') {
    // Otherwise treat it as an attribute if this is not a function or an object.
    script.setAttribute(key, value);
  }
}
```

For reference this is the exact technique used by any other vdom libraries out there. Fair warning 😅  I wrote it by hand so there might be some execution errors in there.

Reply:
Huh, interesting. :thinking:

I thought that using `element.setAttribute()` would handle that case just fine. I mean, if you run

```js
script.setAttribute("onLoad", () => console.log("script loaded!));
```

you're actually setting the `onload` attribute, not the `onLoad` one.

Or is this something that doesn't work the same for all browsers/versions?

Reply:
Unfortunately `.setAttribute` would stringify the value of the attribute. So the code above would actually become:

```html
<script onload="() => console.log('script loaded!')"></script>
```

Which is not what we want to achieve. We want to add a `onload` listener instead on the element DOM reference.

Reply:
Oh, I see, thanks for the clarification. :blush: :+1:

Reply:
Thanks for the helps and suggestions!
Now `className`, `onLoad` and similars are working. Data attributes can be set like `data-value='3'`
An example:
```
<Script
    id='myScript'
    data-value='3'
    onLoad={() => {console.log('Scipt Loaded')}}
    className='coolScript'
    defer=''
    crossOrigin='anonymous'
    src='//some.cdn.com/some/script.js'
  />`
```

Reply:
Could you please add unit tests for all those props? 😄

- Some random prop.
- The `className` prop.
- Data attributes.
- The `onLoad` prop.
- ...

Those tests are really simple. We use something like this:

```js
test("should work with data attributes", () => {
  const ScriptTest = create(<Script data-value="3" />);
  expect(ScriptTest.toJSON()).toMatchSnapshot();
});
```

You can use this tests of `<Switch>` as a reference to set them up: https://github.com/frontity/frontity/blob/dev/packages/components/__tests__/switch.tests.tsx

Thanks!!

Reply:
I tried to create the test cases, but in the .snap file all the cases are `null`.
I could not figure it out, why.

Reply:
How can I test, if <script> tag is added to body? react-test-renderer `create` will not work, as the <Script> component return null.

Reply:
Ohh, yes, that's right.

I've moved them to the e2e tests and they pass 🙂 (https://github.com/frontity/frontity/pull/770/commits/75553ade3d1e052d1443203565bfb7c5180565ee).

Reply:
Oh, I removed the `onLoad` test. As we are creating the `<script>` tags after hydration I don't think it makes sense. And I guess that if someone wants to use it they can always use `<script>` instead of `<Script>`.